proposals/proposal_1/lib/src/logging/logging_service.dart

```diff
@@ -0,0 +1,25 @@
+/// A Service that records application messages or exceptions.
+///
+/// This class should be used throughout the application in place of print or
+/// debugPrint statements.
+///
+/// There are three implementations of this abstract class:
+///   - Release version: Sends records to a logging or error capturing service,
+///     such as Firebase Crashlytics or Sentry. Modify this file to change where
+///     logs are sent in release builds.
+///   - Debug version: Prints information to the console as you work. You
+///     should only modify this file if you want to change how to print messages
+///     during development.
+///   - Profile version: Does nothing. Printing or sending logs to a service may
+///     cause unwanted jank during a performance profile. You should only change
+///     this file if you want to modify how logs are handled during performance
+///     profiles.
```

Comment:
I usually disable printing in profile mode, since it adds noise to the benchmark results. 

I also disable reporting to a service in profile mode, since profiles may run lots and lots of times on CI servers and you do not want to send that data to Crashlytics or Sentry.